R/redundant_ifelse_linter.R

```diff
@@ -46,17 +46,17 @@ redundant_ifelse_linter <- function(allow10 = FALSE) {
     if (!allow10) {
       num_expr <- xml2::xml_find_all(xml, num_xpath)
       matched_call <- xp_call_name(num_expr)
-      # [1] call; [2] logical condiditon
+      # [1] call; [2] logical condition
       first_arg <- xml2::xml_find_chr(num_expr, "string(expr[3]/NUM_CONST)")
       second_arg <- xml2::xml_find_chr(num_expr, "string(expr[4]/NUM_CONST)")
+      replacement_numeric <- ifelse(first_arg == "0" & second_arg == "1", "as.numeric(!x)", "as.numeric(x)")
       replacement <- ifelse(
         first_arg %in% c("0", "1") | second_arg %in% c("0", "1"),
-        "as.numeric",
-        "as.integer"
+        replacement_numeric,
+        "as.integer(x)"
```

Comment:
shouldn't ifelse(x, 0L, 1L) also recommend negation?

sorry for all the back & forth here...

Reply:
Ah, sorry, missed that case here. Yes, we should also recommend negation here.

It should be `as.integer(!x)`, right? Since the original conditions are `0L` and `1L`?

``` r
x <- 8

ifelse(x > 5, 0L, 1L)
#> [1] 0

as.numeric(! (x > 5))
#> [1] 0
as.integer(! (x > 5))
#> [1] 0
```

<sup>Created on 2022-06-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
right, exactly